### PR TITLE
Add dcm2niix

### PR DESCRIPTION
**Pull request by kastman**
OSX and linux should work; testing plus prelim Win test

### CODE REVIEW

**Comment by kastman** [comment]
Would @neurolabusc be interested in being a maintainer as well?

**Comment by neurolabusc** [comment]
@kastman I am not very familiar with conda or python. I would be happy to help if you identify issues, but I do not think I am the appropriate point man for this. Another option is @ningfei who created the superbuild for dcm2niix (as well as many contributions to this project and others).

**Comment by kastman** [comment]
Hi @neurolabusc and @ningfei, I don't anticipate any issues as I'm signing up to update when new releases come out; this is mostly so that you have control of the repo if you need it, and get pinged if there are any hiccups just so you're aware of them. 

In my experience conda is pretty straightforward and doesn't require any python knowledge; this is doubling up a little bit on the superbuild but makes for a one-line install (and easy one-line dependency in other projects). @ningfei, do you want to be included too?

**Comment by neurolabusc** [comment]
OK, sign me up.

**Comment by ningfei** [comment]
Hi @kastman, not familiar with conda-forge but happy to sign up!

**Comment by ningfei** [comment]
Thanks!

**Comment by kastman** [comment]
Two last things:

1. Note that this currently omits support for JPEG2000, as the help sounded like it was pretty sparsely used. Is this something you would want in a release, or is it alright to not be included?

2. I looked for a release announce mailing list to join, but don't see any. I'll try to keep my eyes open for releases, but feel free to ping me when they come out and I'll update the feedstock (dcm2niix's own repo on conda-forge). Thanks!

**Comment by neurolabusc** [comment]
1. I agree. As noted in the dcm2niix COMPILE.MD notes, JPEG2000 did not see widespread adoption. Each of the two open source libraries has issues. Jasper has problems with the high precision seen in medical images. The OpenJPEG library changed and at one stage they added a kludge for defective Adobe images that must be disabled (requiring recompile) for some legitimate images. I have only seen JPEG2000 created by two proprietary products: the proprietary DCMTK DCMJP2K and Osirix. I would strongly discourage anyone from using this format as it may wreak havoc for future archival work. Having said that, @ningfei did an amazing job developing a cmake script for dcm2niix that binds to OpenJPEG on many platforms. Indeed, the AppVeyor and Travis scripts on the dcm2niix main page automatically recompile to MacOS, Linux and Windows after each commit. 

2. If there is an easy way to create a release announce mailing list, tell me.

**Comment by kastman** [comment]
Sounds good; as someone who often does the exact same thing, the effort to make that cmake script is definitely appreciated! But I'm happy not incorporating it. Sorry!

The only list I was thinking of was the NITRC, but it looks like that's possibly tied to their own cvs repository? I was thinking something along the lines of an existing google group, but don't see one.  I'll just check in periodically,

**Comment by kastman** [comment]
@conda-forge/statuspage Ready to merge?

**Comment by jakirkham** [comment]
Think you mean @conda-forge/staged-recipes ;)

**Comment by kastman** [comment]
Sorry for the spam - it was a browser autocorrect but I got it already. Thanks!

File: recipes/dcm2niix/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url:  https://github.com/rordenlab/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (jakirkham) [inline recipes/dcm2niix/meta.yaml]: Should this be skipped? Was just looking at the `bld.bat` above and it seemed mostly ok. :/

**Comment by neurolabusc** [comment]
Just as a comment,  dcm2niix v1.0.2018032 is a pre-release 
  https://github.com/rordenlab/dcm2niix/releases
I hope to close the final issues within the next week, once we get validation datasets for Philips 
  https://github.com/rordenlab/dcm2niix/issues/182
I personally think v1.0.2018032 is better than any prior stable releases. This would be a great time to test it for any unintended consequences.

**Comment by kastman** [comment]
Wouldn't hurt to build the previous release as well; I'll probably add the release once it's been pulled into conda-forge as a feedstock officially.

The Windows build failed, but I'll have to look at that in the morning as I'm not as familiar with that. I thought the whole point of the toolchain dep was to work cross-platform?

**Comment by jakirkham** [comment]
In that case, would recommend starting with the older release here and then updating the feedstock. That comes with the benefit that history in the feedstock is linear and corresponds to the version update order.

**Comment by kastman** [comment]
Creating and using a `build` directory like `build.sh` was all that was needed for the windows `build.bat`. Should be really ready now!

**Comment by dougalsutherland** [comment]
Should `pigz` be a `run` requirement too?

> `Compression will be faster with 'pigz' installed`

**Comment by neurolabusc** [comment]
TL;DR Pigz is optional, I leave it to your discretion.

Pigz can accelerate creation of GZ compressed images. If pigz is not found, compression uses the internal single-threaded compressor. If pigz is found, then dcm2niix will use parallel processing pigz if the user specifies '-z y' and the internal compressor if the user specifies '-z i'. For most users, pigz is much faster, however since it requires first saving the uncompressed data to disk and then loading and compressing it, it can be slower for systems with poor disk I/O (e.g. clusters).

**Comment by kastman** [comment]
In this case, we probably want to leave pigz off as a required dependency then, and we can suggest people install it themselves if they want. Conda has no method for recommendations, right, like pip does? Wondering if I should also take it out of the build environment as well?

**Comment by neurolabusc** [comment]
Well, dcm2niix generates this suggestion whenever you create a gz output without pigz available. Therefore, I think the software generates this hint in its output. It is certainly an option, not a requirement.

**Comment by jakirkham** [comment]
Could you please try pushing again @kastman. Seems CircleCI doesn't like restarting this one for some reason. 😕

**Comment by kastman** [comment]
That's odd. Force-pushed with an empty amend commit (I think that's the right way to do it?)

**Comment by jakirkham** [comment]
Thanks. Let me take a closer look.

**Comment by kastman** [comment]
@jakirkham Thanks for fixing this on master. However, although your new change fixes circle, it broke appveyor, which was wokring directly before the new master merge. 

What was the change? Any chance you could check that out? Thanks!

**Comment by neurolabusc** [comment]
I noticed that appveyor is failing as it uses an old version of the microsoft compiler. @ningfei suggested we detect Visual Studio versions prior to 2013  with `#if _MSC_VER ​< 1800` conditionals. I have updated the dcm2niix master code, but do not have an old version of Visual Studio, so perhaps someone wants to test the latest code and see if it works.

**Comment by kastman** [comment]
Whoops, looks like @neurolabusc may have just pushed a fix to the upstream cmake files. I'll take a look. Not sure how to specify it, but should we be using a newer version of Visual Studio?

**Comment by kastman** [comment]
Testing the newest code will work going forward, but the conda paradigm is to grab existing tarballs, not `git clone`, (though I can switch to test, of course). If there's no reason to use the older Visual Studio, and it's easy to specify a newer version, I suspect building on conda closer to what you use for real life would be the way to go.

**Comment by neurolabusc** [comment]
The releases tab includes the [latest stable tar.gz](https://github.com/rordenlab/dcm2niix/releases), though this page may include a pre-release as well. I plan to generate a new stable release in the next day or two, so it would be a good time to make sure your process works with the latest code. The enhanced DICOM support means a lot of new changes. Typically, the changes are much more modest.

**Comment by jakirkham** [comment]
Thanks for raising @kastman. Noticed the AppVeyor failure. Though nothing should have changed there recently. Maybe something else is going on?

**Comment by kastman** [comment]
@jakirkham ah, wasn’t aware that the new style compilers were available on conda forge yet; good to know. Any way to tell which version of visual studio they would resolve to, or specify it?

**Comment by ningfei** [comment]
@kastman, also see in the [wiki](https://github.com/conda-forge/staged-recipes/wiki/VC-features) page and windows building [doc](https://conda-forge.org/docs/buildwin.html#building-for-different-vc-versions). It would make things much easier if you can set the compiler to `vc14` as jakirkham [said](https://github.com/conda-forge/ninja-feedstock/blob/5c6cab58934f9382f159099f65598092b79e1fd8/recipe/meta.yaml#L14). As I tested with VS 2008 and VS 2012, there are further issues even with @neurolabusc's latest fix. That's to say, dcm2niix requires VS 2015 to build.

**Comment by kastman** [comment]
I believe this is ready to pull, unless we should switch over to the new style compilers? Could someone from condaforge/core sanity check that and pull? Thanks!

**Comment by kastman** [comment]
@conda-forge/core I think this is ready ready to merge, but it's been a while. Should I re-base against master again, or could someone see if this is ready to go?

As soon as this is pulled in and a feedstock created I'll update with the most recent June release of dcm2niix

**Comment by scopatz** [comment]
It is probably worth merging from master again, I think.

**Comment by neurolabusc** [comment]
If built from the current master, it should report v1.0.20180622
 https://github.com/rordenlab/dcm2niix/releases

**Comment by scopatz** [comment]
Thanks @neurolabusc!

**Comment by kastman** [comment]
Seconded. Thanks @neurolabusc @ningfei @scopatz
